platform/util/ui/src/com/intellij/ui/icons/ImageDescriptor.java

```diff
@@ -26,6 +26,8 @@
 @ApiStatus.Internal
 public final class ImageDescriptor {
   private static final ConcurrentMap<String, Pair<Image, ImageLoader.Dimension2DDouble>> ourCache = ContainerUtil.createConcurrentSoftValueMap();
+  private static final ConcurrentMap<String, Image> ourLargeImageCache = ContainerUtil.createConcurrentWeakValueMap();
+  private static final ConcurrentMap<Image, ImageLoader.Dimension2DDouble> ourLargeImageDimensionMap = ContainerUtil.createConcurrentWeakMap();
```

Comment:
That (second) map contains small objects on weak references. It is a possible case where these small objects are not collected along with the bigger values from the first map. Also, the more weak references are created the harder time would a GC have to deal with that.

It could probably be better to use a small immutable value object to keep both an image and it's dimensions on a hard reference and use the only one map with weak values instead

Reply:
Not sure I understand. The map(s) have to support lookup by a string key. This seems to imply that at least one map has to be keyed by string and contain strong key references. In order for this map not to leak memory, it has to have weak reference values. Weak reference values have to point to objects that are referenced elsewhere while in use. Neither Pair<Image, ImageLoader.Dimension2DDouble>, nor ImageLoader.Dimension2DDouble satisfy this condition. Do you mind elaborating on your proposal?

Reply:
Sorry for the delay in answering. 

I was about to suggest the following:
```
ConcurrentMap<String, Pair<Image, ImageLoader.Dimension2DDouble>> map = ContainerUtil.createConcurrentWeakValueMap();
```
Now I see it will unlikely work because a `Pair<>` object from map's value is only referenced from the weak value map and thus has a good chance to be collected before an Image object is collected itself. Your solution is fine